proms-resources/frontend/src/main/frontend/src/ToUDialog.jsx

```diff
@@ -84,22 +101,54 @@ function ToUDialog(props) {
   }, []);
 
   if (!tou && !error) {
-    return null;
+    return <CircularProgress/>;
```

Comment:
Actually showing a progress here is unnecessary.
```suggestion
    return null;
```